akka-stream/src/main/scala/akka/stream/impl/ActorRefBackpressureSinkActor.scala

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.impl
+
+import akka.actor.{ ActorRef, Props, Status, Terminated }
+import akka.stream.actor.{ ActorSubscriber, ActorSubscriberMessage, RequestStrategy }
+
+/**
+ * INTERNAL API
+ */
+private[akka] object ActorRefBackpressureSinkActor {
+  def props(ref: ActorRef, maxBuffer: Int, onInitMessage: Any, onCompleteMessage: Any): Props =
+    Props(new ActorRefBackpressureSinkActor(ref, maxBuffer, onInitMessage, onCompleteMessage))
+}
+
+/**
+ * INTERNAL API
+ */
+private[akka] class ActorRefBackpressureSinkActor(ref: ActorRef, maxBuffer: Int, onInitMessage: Any,
+                                                  onCompleteMessage: Any) extends ActorSubscriber {
+  import ActorSubscriberMessage._
+  var buffer: Vector[Any] = Vector.empty
+
+  var backpressureReceived = false
```

Comment:
please rename: what is received is not backpressure but a confirmation or acknowledgement